azure_arc_k8s_jumpstart/aksengine/arc_k8s_aksengine.azcli

```diff
@@ -0,0 +1,432 @@
+# Deploy aks-engine cluster #
+#############################
+
+# Variables
+k8s_rg=aksengine
+location=westeurope
+arc_rg=k8sarc
+arc_name=myaksengine
+
+# Create RG for AKS engine cluster
+az group create -n $k8s_rg -l $location
+
+# Install AKS engine
+# Go here: https://github.com/Azure/aks-engine/releases/latest
+# Example with v0.61.0:
+aksengine_exec=$(which aks-engine)
+if [[ -n "$aksengine_exec" ]]
+then
+    echo "aks-engine executable found in ${aksengine_exec}"
+else
+    echo "Downloading and installing aks-engine executable..."
+    aksengine_tmp=/tmp/aksengine.tar.gz
+    wget https://github.com/Azure/aks-engine/releases/download/v0.61.0/aks-engine-v0.61.0-linux-amd64.tar.gz -O $aksengine_tmp
+    tar xfvz $aksengine_tmp -C /tmp/
+    sudo cp /tmp/aks-engine-v0.61.0-linux-amd64/aks-engine /usr/local/bin
+fi
+
+# Retrieve Service Principal form your AKV, required for AKS engine
+purpose=aksengine
+keyvault_appid_secret_name=$purpose-sp-appid
+keyvault_password_secret_name=$purpose-sp-secret
+keyvault_name=erjositoKeyvault
+keyvault_appid_secret_name=$purpose-sp-appid
+keyvault_password_secret_name=$purpose-sp-secret
+sp_app_id=$(az keyvault secret show --vault-name $keyvault_name -n $keyvault_appid_secret_name --query 'value' -o tsv 2>/dev/null)
+sp_app_secret=$(az keyvault secret show --vault-name $keyvault_name -n $keyvault_password_secret_name --query 'value' -o tsv 2>/dev/null)
+
+# If they could not be retrieved, generate new ones
+if [[ -z "$sp_app_id" ]] || [[ -z "$sp_app_secret" ]]
+then
+    echo "No SP for AKS-engine could be found in AKV $keyvault_name, generating new ones..."
+    sp_name=$purpose
+    sp_output=$(az ad sp create-for-rbac --name $sp_name --skip-assignment 2>/dev/null)
+    sp_app_id=$(echo $sp_output | jq -r '.appId')
+    sp_app_secret=$(echo $sp_output | jq -r '.password')
+    # Store the created app ID and secret in an AKV
+    az keyvault secret set --vault-name $keyvault_name -n $keyvault_appid_secret_name --value $sp_app_id
+    az keyvault secret set --vault-name $keyvault_name -n $keyvault_password_secret_name --value $sp_app_secret
+else
+    echo "Service Principal $sp_app_id and secret successfully retrieved from AKV $keyvault_name"
+fi
+
+# Grant access to the SP to the new RG
+scope=$(az group show -n $k8s_rg --query id -o tsv)
+assignee=$(az ad sp show --id $sp_app_id --query objectId -o tsv)
+az role assignment create --scope $scope --role Contributor --assignee $assignee
+
+# File containing the description of the AKS engine cluster to create
+aksengine_cluster_file="/tmp/aksengine_cluster.json" 
+
+# Option1: Retrieve example JSON file describing a basic cluster
+url=https://raw.githubusercontent.com/Azure/aks-engine/master/examples/kubernetes.json
+wget $url -O $aksengine_cluster_file
+# You can modify the kubernetes.json file, for example with smaller VM sizes such as Standard_B2ms
+
+# Option 2: Create a cluster file from scratch:
+aksengine_vm_size=Standard_B2ms
```

Comment:
Move all variables at the top of the script so it is easier to customize

Reply:
Done!